lesson_13/mosut_sorin/hw_lesson13_inheritance_mosut_sorin.py

```diff
@@ -0,0 +1,118 @@
+"""
+# Lesson 13 - Inheritance
+
+
+We have new customers for our Polygons company.
+
+    * They need to create square objects with a certain area
+
+    * They need a method to compute the perimeter only for triangle objects
+
+Requirements:
+
+    * add an alternative constructor to Square class that takes the area as an
+argument and creates a square object with the apropriate sides
+
+> example :
+>
+>     >> sq = Square.from_area(4)
+>     >> print(sq)
+>     >> Side 1 with lenght: 2
+>     >> Side 2 with lenght: 2
+>     >> Side 3 with lenght: 2
+>     >> Side 4 with lenght: 2
+
+
+    * we want to make our perimeter method also available to other shapes in the future. Create a mixin
+
+class for perimeter that contains the perimeter method
+"""
+
+from math import sqrt
+
+
+class PerimeterMixin:
+
+    def perimeter(self):
+        return f'The perimeter of a {self.geometric_figure} with sides {self.sides} is {sum(self.sides):.2f}\n'
+
+
+class Polygons(PerimeterMixin):
```

Comment:
The requirement of the challenge was to have the perimeter method available only for triangle class. If we use it like this we will have perimeter method on all the shapes that use Polygons as a parent